Vizio coordinator update test is flaky

### The problem

I've seen this test failing multiple times on unrelated PRs and it requires restarting the whole CI to pass all jobs.

Test name: `tests/components/vizio/test_init.py::test_coordinator_update_failure`

Example which failed on `dev` branch: https://github.com/home-assistant/core/runs/2594855564

I think this test should be fixed or removed.

cc @raman325 

### What is version of Home Assistant Core has the issue?

dev

### What was the last working version of Home Assistant Core?

_No response_

### What type of installation are you running?

Home Assistant Core

### Integration causing the issue

Vizio

### Link to integration documentation on our website

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```
        async_fire_time_changed(hass, now + timedelta(days=10))
        await hass.async_block_till_done()
>       assert "Unable to retrieve the apps list from the external server" in caplog.text
E       AssertionError: assert 'Unable to retrieve the apps list from the external server' in 'INFO     homeassistant.loader:loader.py:344 Loaded vizio from homeassistant.components.vizio\nINFO     homeassistant....nds\nDEBUG    homeassistant.components.vizio:update_coordinator.py:253 Finished fetching vizio data in 0.000 seconds\n'
E        +  where 'INFO     homeassistant.loader:loader.py:344 Loaded vizio from homeassistant.components.vizio\nINFO     homeassistant....nds\nDEBUG    homeassistant.components.vizio:update_coordinator.py:253 Finished fetching vizio data in 0.000 seconds\n' = <_pytest.logging.LogCaptureFixture object at 0x7fb8458e2910>.text
```

### Additional information

_No response_

--- a/homeassistant/components/vizio/__init__.py
+++ b/homeassistant/components/vizio/__init__.py
@@ -113,6 +113,9 @@ class VizioAppsDataUpdateCoordinator(DataUpdateCoordinator):
         """Update data via library."""
         data = await gen_apps_list_from_url(session=async_get_clientsession(self.hass))
         if not data:
+            # For every failure, increase the fail count until we reach the threshold.
+            # We then log a warning, increase the threshold, and reset the fail count.
+            # This is here to prevent silent failures but to reduce repeat logs.
             if self.fail_count == self.fail_threshold:
                 _LOGGER.warning(
                     (
@@ -126,6 +129,7 @@ class VizioAppsDataUpdateCoordinator(DataUpdateCoordinator):
             else:
                 self.fail_count += 1
             return self.data
+        # Reset the fail count and threshold when the data is successfully retrieved
         self.fail_count = 0
         self.fail_threshold = 10
         return sorted(data, key=lambda app: app["name"])

--- a/tests/components/vizio/test_init.py
+++ b/tests/components/vizio/test_init.py
@@ -94,14 +94,11 @@ async def test_coordinator_update_failure(
     assert len(hass.states.async_entity_ids(MP_DOMAIN)) == 1
     assert DOMAIN in hass.data
 
-    for days in range(1, 10):
+    # Failing 25 days in a row should result in a single log message
+    # (first one after 10 days, next one would be at 30 days)
+    for days in range(1, 25):
         async_fire_time_changed(hass, now + timedelta(days=days))
         await hass.async_block_till_done()
-        assert (
-            "Unable to retrieve the apps list from the external server"
-            not in caplog.text
-        )
 
-    async_fire_time_changed(hass, now + timedelta(days=10))
-    await hass.async_block_till_done()
-    assert "Unable to retrieve the apps list from the external server" in caplog.text
+    err_msg = "Unable to retrieve the apps list from the external server"
+    assert len([record for record in caplog.records if err_msg in record.msg]) == 1

